cohort-engine/src/main/java/com/ibm/cohort/engine/measure/CDMMeasureEvaluation.java

```diff
@@ -197,6 +208,58 @@ protected static void addDefineEvaluationToReport(MeasureReport report, CDMConte
 			}
 		}
 	}
+	
+	protected static void addParametersToReport(MeasureReport report, Measure measure, Context context, Map<String, Parameter> parameterMap) {
+		Set<String> parameterNames = new HashSet<>();
+		
+		// Check for special parameters we handle elsewhere
+		if (context.resolveParameterRef(null, CDMConstants.MEASUREMENT_PERIOD) != null) {
+			parameterNames.add(CDMConstants.MEASUREMENT_PERIOD);
+		}
+		
+		if (context.resolveParameterRef(null, CDMConstants.PRODUCT_LINE) != null) {
+			parameterNames.add(CDMConstants.PRODUCT_LINE);
+		}
+
+		if (parameterMap != null) {
+			parameterNames.addAll(parameterMap.keySet());
+		}
+		
+		List<Extension> parameterExtensions = measure.getExtensionsByUrl(CDMConstants.MEASURE_PARAMETER_URL);
+		for (Extension e : parameterExtensions) {
+			ParameterDefinition parameterDefinition = (ParameterDefinition) e.getValue();
+			parameterNames.add(parameterDefinition.getName());
+		}
+		
+		parameterNames.forEach(x -> report.addExtension(createParameterExtension(context, x)));
```

Comment:
Pedandic:  Would it make more sense to have this method return `List<Extension>`, and then add the new extensions to the report in the calling method?  I like to avoid side effecting when possible, but this isn't required in the slightest.